### ISSUE TITLE
Microsoft.UI.Xaml 2.8.6 install hangs

### ISSUE THREAD

**Issue by agiacomolli** [microsoft/microsoft-ui-xaml#10183]
### Describe the bug

The install hangs without any output using `Add-AppxProvisionedPackage`.

### Steps to reproduce the bug

```
Add-AppxProvisionedPackage -Online -PackagePath .\Microsoft.UI.Xaml.2.8.x64.appx -SkipLicense -LogLevel WarningsInfo -LogPath c:\dism.log
```

### Expected behavior

The package being correctly provisioned.

### NuGet package version

None

### Additional context

Package version: 2.8.6
Windows version: 11 (10.0.21996)

Logs mention missing dependencies. Full DISM log:

```
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll:                                            - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: <----- Starting DismApi.dll session -----> - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll:                                            - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: Host machine information: OS Version=10.0.21996, Running architecture=amd64, Number of processors=4 - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: API Version 10.0.21996.1 - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: Parent process command line: powershell - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Input parameters: LogLevel: 2, LogFilePath: c:\dism.log, ScratchDirectory: (null) - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Initialized GlobalConfig - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Initialized SessionTable - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Lookup in table by path failed for: DummyPath-2BA51B78-C7F7-4910-B99D-BB7345357CDC - CTransactionalImageTable::LookupImagePath
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Waiting for m_pInternalThread to start - CCommandThread::Start
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=9008 Enter CCommandThread::CommandThreadProcedureStub - CCommandThread::CommandThreadProcedureStub
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=9008 Enter CCommandThread::ExecuteLoop - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 CommandThread StartupEvent signaled - CCommandThread::WaitForStartup
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 m_pInternalThread started - CCommandThread::Start
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Created g_internalDismSession - DismInitializeInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Input parameters: ImagePath: DISM_{53BFAE52-B167-4E2F-A258-0A37B57FF845}, WindowsDirectory: (null), SystemDrive: (null) - DismOpenSessionInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Lookup in table by path failed for: DRIVE_C - CTransactionalImageTable::LookupImagePath
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Waiting for m_pInternalThread to start - CCommandThread::Start
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Enter CCommandThread::CommandThreadProcedureStub - CCommandThread::CommandThreadProcedureStub
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Enter CCommandThread::ExecuteLoop - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 CommandThread StartupEvent signaled - CCommandThread::WaitForStartup
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 m_pInternalThread started - CCommandThread::Start
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:17, Info                  DISM   PID=9832 TID=7232 Scratch directory set to 'C:\Windows\TEMP\'. - CDISMManager::put_ScratchDir
2024-11-19 06:25:17, Info                  DISM   PID=9832 TID=7232 DismCore.dll version: 10.0.21996.1 - CDISMManager::FinalConstruct
2024-11-19 06:25:17, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:17, Info                  DISM   PID=9832 TID=7232 Successfully loaded the ImageSession at "C:\Windows\SYSTEM32\Dism" - CDISMManager::LoadLocalImageSession
2024-11-19 06:25:17, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:17, Info                  DISM   DISM Provider Store: PID=9832 TID=7232 Found and Initialized the DISM Logger. - CDISMProviderStore::Internal_InitializeLogger
2024-11-19 06:25:17, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 Successfully created the local image session and provider store. - CDISMManager::CreateLocalImageSession
2024-11-19 06:25:17, Info                  DISM   DISM FFU Provider: PID=9832 TID=7232 [C:\] is not recognized by the DISM FFU provider. - CFfuImage::Initialize
2024-11-19 06:25:17, Info                  DISM   DISM Imaging Provider: PID=9832 TID=7232 The provider FfuManager does not support CreateDismImage on C:\ - CGenericImagingManager::CreateDismImage
2024-11-19 06:25:17, Info                  DISM   DISM VHD Provider: PID=9832 TID=7232 [C:\] is not recognized by the DISM VHD provider. - CVhdImage::Initialize
2024-11-19 06:25:17, Info                  DISM   DISM Imaging Provider: PID=9832 TID=7232 The provider VHDManager does not support CreateDismImage on C:\ - CGenericImagingManager::CreateDismImage
[9832.7232] [0x80070002] FIOReadFileIntoBuffer:(1456): The system cannot find the file specified.
[9832.7232] [0xc142011c] UnmarshallImageHandleFromDirectory:(641)
[9832.7232] [0xc142011c] WIMGetMountedImageHandle:(2910)
2024-11-19 06:25:17, Info                  DISM   DISM WIM Provider: PID=9832 TID=7232 [C:\] is not a WIM mount point. - CWimMountedImageInfo::Initialize
2024-11-19 06:25:17, Info                  DISM   DISM Imaging Provider: PID=9832 TID=7232 The provider WimManager does not support CreateDismImage on C:\ - CGenericImagingManager::CreateDismImage
2024-11-19 06:25:17, Info                  DISM   DISM Imaging Provider: PID=9832 TID=7232 No imaging provider supported CreateDismImage for this path - CGenericImagingManager::CreateDismImage
[9832.7232] [0x80070002] FIOReadFileIntoBuffer:(1456): The system cannot find the file specified.
[9832.7232] [0xc142011c] UnmarshallImageHandleFromDirectory:(641)
[9832.7232] [0xc142011c] WIMGetMountedImageHandle:(2910)
2024-11-19 06:25:17, Info                  DISM   DISM WIM Provider: PID=9832 TID=7232 [C:\] is not a WIM mount point. - CWimMountedImageInfo::Initialize
2024-11-19 06:25:17, Info                  DISM   DISM FFU Provider: PID=9832 TID=7232 [C:\] is not recognized by the DISM FFU provider. - CFfuImage::Initialize
2024-11-19 06:25:17, Info                  DISM   DISM VHD Provider: PID=9832 TID=7232 [C:\] is not recognized by the DISM VHD provider. - CVhdImage::Initialize
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 physical location path: C:\ - CDISMManager::CreateImageSession
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 Event name for current DISM session is Global\{796F4D68-7657-4A1D-B66D-B0E71F17752C} - CDISMManager::CheckSessionAndLock
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 Create session event 0xda0 for current DISM session and event name is Global\{796F4D68-7657-4A1D-B66D-B0E71F17752C}  - CDISMManager::CheckSessionAndLock
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 Copying DISM from "C:\Windows\System32\Dism" - CDISMManager::CreateImageSessionFromLocation
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 Successfully loaded the ImageSession at "C:\Windows\TEMP\391DE563-ECBD-4B40-857A-7CE77BBF166C" - CDISMManager::LoadRemoteImageSession
2024-11-19 06:25:17, Info                  DISM   DISM Image Session: PID=5720 TID=9524 Instantiating the Provider Store. - CDISMImageSession::get_ProviderStore
2024-11-19 06:25:17, Info                  DISM   DISM OS Provider: PID=5720 TID=9524 Defaulting SystemPath to C:\ - CDISMOSServiceManager::Final_OnConnect
2024-11-19 06:25:17, Info                  DISM   DISM OS Provider: PID=5720 TID=9524 Defaulting Windows folder to C:\Windows - CDISMOSServiceManager::Final_OnConnect
2024-11-19 06:25:17, Info                  DISM   DISM Provider Store: PID=5720 TID=9524 Attempting to initialize the logger from the Image Session. - CDISMProviderStore::Final_OnConnect
2024-11-19 06:25:17, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:17, Info                  DISM   DISM Provider Store: PID=5720 TID=9524 Found and Initialized the DISM Logger. - CDISMProviderStore::Internal_InitializeLogger
2024-11-19 06:25:17, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:17, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 Image session successfully loaded from the temporary location: C:\Windows\TEMP\391DE563-ECBD-4B40-857A-7CE77BBF166C - CDISMManager::CreateImageSession
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Target image information: OS Version=10.0.21996.1, Image architecture=amd64 - CDismCore::LogImageSessionDetails
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Session id is: 2 - DismOpenSessionInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Input parameters: Session: 2 - DismGetOsInfoInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Initialized default cancel event - DismGetOsInfoInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Input parameters: 
 	Session: 2, 
 	AppPath: .\Microsoft.UI.Xaml.2.8.x64.appx, 
 	DependencyPackageCount: 0, 
 	OptionalPackageCount: 0, 
 	LicensePathCount: 0, 
 	CustomDataPath: (null), 
 	SkipLicense: 1 - DismAddProvisionedAppxPackageInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Initialized default cancel event - DismAddProvisionedAppxPackageInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Calling into AddAllUserAppxPackage with mainPackagePath = .\Microsoft.UI.Xaml.2.8.x64.appx, CustomDataPath = (null), SkipLicense = -1, Regions = (null) - CAddAppxCommandObject::InternalExecute
2024-11-19 06:25:17, Info                  DISM   DISM OS Provider: PID=5720 TID=9524 Determined System directory to be C:\Windows\System32 - CDISMOSServiceManager::get_SystemDirectory
2024-11-19 06:25:17, Error                 DISM   DISM Appx Provider: PID=5720 TID=9524 Failed to create AppX Provider session lock 'Global\DismAppxProviderOnline{4E349A62-112B-444A-946F-56F1ABB80762}' because another online session is processing the image - CAppxManager::AcquireSessionLockIfOnline(hr:0xc1570106)
2024-11-19 06:25:17, Error                 DISM   DISM Appx Provider: PID=5720 TID=9524 Failed while acquiring Appx Provider session lock - CAppxManager::AddAllUserAppxPackageWithOptionalPackagesAndRegionAndStubPackageOption(hr:0xc1570106)
2024-11-19 06:25:17, Error                 DISM   API: PID=9832 TID=7232 Call to AddAllUserAppxPackage failed with hr = 0xc1570106 - CAddAppxCommandObject::InternalExecute(hr:0xc1570106)
2024-11-19 06:25:17, Error                 DISM   API: PID=9832 TID=7232 InternalExecute failed - CBaseCommandObject::Execute(hr:0xc1570106)
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Input parameters: Session: 2 - DismCloseSessionInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 GetReferenceCount hr: 0x0 - CSessionTable::RemoveSession
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Refcount for DismSession= 2s 0 - CSessionTable::RemoveSession
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 ExecuteLoop: Cancel signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Leave CCommandThread::ExecuteLoop - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   DISM Manager: PID=9832 TID=7232 Closing session event handle 0xda0 - CDISMManager::CleanupImageSessionEntry
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=7232 Leave CCommandThread::CommandThreadProcedureStub - CCommandThread::CommandThreadProcedureStub
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 GetReferenceCount hr: 0x0 - CSessionTable::RemoveSession
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Refcount for DismSession= 1s 0 - CSessionTable::RemoveSession
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=9008 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=9008 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=9008 ExecuteLoop: Cancel signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=9008 Leave CCommandThread::ExecuteLoop - CCommandThread::ExecuteLoop
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=9008 Leave CCommandThread::CommandThreadProcedureStub - CCommandThread::CommandThreadProcedureStub
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Deleted g_internalDismSession - DismShutdownInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 Shutdown SessionTable - DismShutdownInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll:                                          - DismShutdownInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: <----- Ending DismApi.dll session -----> - DismShutdownInternal
2024-11-19 06:25:17, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll:                                          - DismShutdownInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll:                                            - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: <----- Starting DismApi.dll session -----> - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll:                                            - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: Host machine information: OS Version=10.0.21996, Running architecture=amd64, Number of processors=4 - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: API Version 10.0.21996.1 - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 DismApi.dll: Parent process command line: powershell - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 Input parameters: LogLevel: 2, LogFilePath: c:\dism.log, ScratchDirectory: (null) - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 Initialized GlobalConfig - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 Initialized SessionTable - DismInitializeInternal
2024-11-19 06:25:27, Info                  DISM   API: PID=9832 TID=4488 Lookup in table by path failed for: DummyPath-2BA51B78-C7F7-4910-B99D-BB7345357CDC - CTransactionalImageTable::LookupImagePath
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Waiting for m_pInternalThread to start - CCommandThread::Start
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=3952 Enter CCommandThread::CommandThreadProcedureStub - CCommandThread::CommandThreadProcedureStub
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=3952 Enter CCommandThread::ExecuteLoop - CCommandThread::ExecuteLoop
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 CommandThread StartupEvent signaled - CCommandThread::WaitForStartup
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 m_pInternalThread started - CCommandThread::Start
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Created g_internalDismSession - DismInitializeInternal
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Input parameters: ImagePath: DISM_{53BFAE52-B167-4E2F-A258-0A37B57FF845}, WindowsDirectory: (null), SystemDrive: (null) - DismOpenSessionInternal
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Lookup in table by path failed for: DRIVE_C - CTransactionalImageTable::LookupImagePath
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Waiting for m_pInternalThread to start - CCommandThread::Start
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 Enter CCommandThread::CommandThreadProcedureStub - CCommandThread::CommandThreadProcedureStub
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 Enter CCommandThread::ExecuteLoop - CCommandThread::ExecuteLoop
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 CommandThread StartupEvent signaled - CCommandThread::WaitForStartup
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 m_pInternalThread started - CCommandThread::Start
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:28, Info                  DISM   PID=9832 TID=9420 Scratch directory set to 'C:\Windows\TEMP\'. - CDISMManager::put_ScratchDir
2024-11-19 06:25:28, Info                  DISM   PID=9832 TID=9420 DismCore.dll version: 10.0.21996.1 - CDISMManager::FinalConstruct
2024-11-19 06:25:28, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:28, Info                  DISM   PID=9832 TID=9420 Successfully loaded the ImageSession at "C:\Windows\SYSTEM32\Dism" - CDISMManager::LoadLocalImageSession
2024-11-19 06:25:28, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:28, Info                  DISM   DISM Provider Store: PID=9832 TID=9420 Found and Initialized the DISM Logger. - CDISMProviderStore::Internal_InitializeLogger
2024-11-19 06:25:28, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:28, Info                  DISM   DISM Manager: PID=9832 TID=9420 Successfully created the local image session and provider store. - CDISMManager::CreateLocalImageSession
2024-11-19 06:25:28, Info                  DISM   DISM FFU Provider: PID=9832 TID=9420 [C:\] is not recognized by the DISM FFU provider. - CFfuImage::Initialize
2024-11-19 06:25:28, Info                  DISM   DISM Imaging Provider: PID=9832 TID=9420 The provider FfuManager does not support CreateDismImage on C:\ - CGenericImagingManager::CreateDismImage
2024-11-19 06:25:28, Info                  DISM   DISM VHD Provider: PID=9832 TID=9420 [C:\] is not recognized by the DISM VHD provider. - CVhdImage::Initialize
2024-11-19 06:25:28, Info                  DISM   DISM Imaging Provider: PID=9832 TID=9420 The provider VHDManager does not support CreateDismImage on C:\ - CGenericImagingManager::CreateDismImage
[9832.9420] [0x80070002] FIOReadFileIntoBuffer:(1456): The system cannot find the file specified.
[9832.9420] [0xc142011c] UnmarshallImageHandleFromDirectory:(641)
[9832.9420] [0xc142011c] WIMGetMountedImageHandle:(2910)
2024-11-19 06:25:28, Info                  DISM   DISM WIM Provider: PID=9832 TID=9420 [C:\] is not a WIM mount point. - CWimMountedImageInfo::Initialize
2024-11-19 06:25:28, Info                  DISM   DISM Imaging Provider: PID=9832 TID=9420 The provider WimManager does not support CreateDismImage on C:\ - CGenericImagingManager::CreateDismImage
2024-11-19 06:25:28, Info                  DISM   DISM Imaging Provider: PID=9832 TID=9420 No imaging provider supported CreateDismImage for this path - CGenericImagingManager::CreateDismImage
[9832.9420] [0x80070002] FIOReadFileIntoBuffer:(1456): The system cannot find the file specified.
[9832.9420] [0xc142011c] UnmarshallImageHandleFromDirectory:(641)
[9832.9420] [0xc142011c] WIMGetMountedImageHandle:(2910)
2024-11-19 06:25:28, Info                  DISM   DISM WIM Provider: PID=9832 TID=9420 [C:\] is not a WIM mount point. - CWimMountedImageInfo::Initialize
2024-11-19 06:25:28, Info                  DISM   DISM FFU Provider: PID=9832 TID=9420 [C:\] is not recognized by the DISM FFU provider. - CFfuImage::Initialize
2024-11-19 06:25:28, Info                  DISM   DISM VHD Provider: PID=9832 TID=9420 [C:\] is not recognized by the DISM VHD provider. - CVhdImage::Initialize
2024-11-19 06:25:28, Info                  DISM   DISM Manager: PID=9832 TID=9420 physical location path: C:\ - CDISMManager::CreateImageSession
2024-11-19 06:25:28, Info                  DISM   DISM Manager: PID=9832 TID=9420 Event name for current DISM session is Global\{EA7E915F-C9DB-411F-BE65-686D08F775FC} - CDISMManager::CheckSessionAndLock
2024-11-19 06:25:28, Info                  DISM   DISM Manager: PID=9832 TID=9420 Create session event 0xde4 for current DISM session and event name is Global\{EA7E915F-C9DB-411F-BE65-686D08F775FC}  - CDISMManager::CheckSessionAndLock
2024-11-19 06:25:28, Info                  DISM   DISM Manager: PID=9832 TID=9420 Copying DISM from "C:\Windows\System32\Dism" - CDISMManager::CreateImageSessionFromLocation
2024-11-19 06:25:28, Info                  DISM   DISM Manager: PID=9832 TID=9420 Successfully loaded the ImageSession at "C:\Windows\TEMP\E9592060-D564-4E21-A30F-A959C69D095E" - CDISMManager::LoadRemoteImageSession
2024-11-19 06:25:28, Info                  DISM   DISM Image Session: PID=8516 TID=8216 Instantiating the Provider Store. - CDISMImageSession::get_ProviderStore
2024-11-19 06:25:28, Info                  DISM   DISM OS Provider: PID=8516 TID=8216 Defaulting SystemPath to C:\ - CDISMOSServiceManager::Final_OnConnect
2024-11-19 06:25:28, Info                  DISM   DISM OS Provider: PID=8516 TID=8216 Defaulting Windows folder to C:\Windows - CDISMOSServiceManager::Final_OnConnect
2024-11-19 06:25:28, Info                  DISM   DISM Provider Store: PID=8516 TID=8216 Attempting to initialize the logger from the Image Session. - CDISMProviderStore::Final_OnConnect
2024-11-19 06:25:28, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:28, Info                  DISM   DISM Provider Store: PID=8516 TID=8216 Found and Initialized the DISM Logger. - CDISMProviderStore::Internal_InitializeLogger
2024-11-19 06:25:28, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:28, Info                  DISM   Initialized Panther logging at c:\dism.log
2024-11-19 06:25:28, Info                  DISM   DISM Manager: PID=9832 TID=9420 Image session successfully loaded from the temporary location: C:\Windows\TEMP\E9592060-D564-4E21-A30F-A959C69D095E - CDISMManager::CreateImageSession
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 Target image information: OS Version=10.0.21996.1, Image architecture=amd64 - CDismCore::LogImageSessionDetails
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Session id is: 2 - DismOpenSessionInternal
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Input parameters: Session: 2 - DismGetOsInfoInternal
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Initialized default cancel event - DismGetOsInfoInternal
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Input parameters: 
 	Session: 2, 
 	AppPath: .\Microsoft.UI.Xaml.2.8.x64.appx, 
 	DependencyPackageCount: 0, 
 	OptionalPackageCount: 0, 
 	LicensePathCount: 0, 
 	CustomDataPath: (null), 
 	SkipLicense: 1 - DismAddProvisionedAppxPackageInternal
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Initialized default cancel event - DismAddProvisionedAppxPackageInternal
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=4488 Successfully enqueued command object - CCommandThread::EnqueueCommandObject
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 ExecuteLoop: CommandQueue signaled - CCommandThread::ExecuteLoop
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 Successfully dequeued command object - CCommandThread::DequeueCommandObject
2024-11-19 06:25:28, Info                  DISM   API: PID=9832 TID=9420 Calling into AddAllUserAppxPackage with mainPackagePath = .\Microsoft.UI.Xaml.2.8.x64.appx, CustomDataPath = (null), SkipLicense = -1, Regions = (null) - CAddAppxCommandObject::InternalExecute
2024-11-19 06:25:28, Info                  DISM   DISM OS Provider: PID=8516 TID=8216 Determined System directory to be C:\Windows\System32 - CDISMOSServiceManager::get_SystemDirectory
2024-11-19 06:25:28, Error                 DISM   DISM Appx Provider: PID=8516 TID=8216 onecore\admin\appmodel\utilities\provisionhelper\msixpackageadapter.cpp(1646)\AppxProvider.dll!00007FFEFA2D16C1: (caller: 00007FFEFA2D0E7A) LogHr(1) tid(2018) 8007029C An assertion failure has occurred.
 - CAppxManager::ResultLoggingCallback
2024-11-19 06:25:28, Error                 DISM   DISM Appx Provider: PID=8516 TID=8216 onecore\admin\appmodel\utilities\provisionhelper\msixpackageadapter.cpp(1646)\AppxProvider.dll!00007FFEFA2D16C1: (caller: 00007FFEFA2D0E7A) LogHr(2) tid(2018) 8007029C An assertion failure has occurred.
    Msg:[No main package dependencies for Microsoft.UI.Xaml.2.8_8.2310.30001.0_x64__8wekyb3d8bbwe, this is a main package and not an optional package.] 
 - CAppxManager::ResultLoggingCallback
```

**Comment by Pinguin2001** [comment]
The build you are running is unsupported, you are running Build 21966, which is a leaked build from May 2021.

Please verify behavoir on a supported build of Windows

**Comment by agiacomolli** [comment]
Thanks @Pinguin2001!

I was not aware that this build is not supported, as the package README mentions "Windows 10 1809 - Build 17763 or newer (including Windows Insider Previews)". Could you point me to where I can find a list of all the supported builds?

We have a wrapper around a tool that has WinUI as a dependency, so being unable to provision WinUI means the end-user cannot use our software. Knowing where we can provision it would allow us to present a better error message to the user.

**Comment by Pinguin2001** [comment]
@agiacomolli No issue here, the operation works perfectly. This command executes the same underlying functionality but via dism directly rather than powershell, which is preferred for such operation because you cannot 100% guarantee ps is present on the users' machine.

```batch
D:\>dism.exe /online /Add-ProvisionedAppxPackage /PackagePath:"D:\Microsoft.UI.Xaml.2.8.x64.appx" /SkipLicense

Deployment Image Servicing and Management tool
Version: 10.0.26100.1150

Image Version: 10.0.26120.2222

The operation completed successfully.
```

Please check if you appx is valid

Via PS it also works perfectly:

```powershell
PS D:\> Add-AppxProvisionedPackage -Online -PackagePath .\Microsoft.UI.Xaml.2.8.x64.appx -SkipLicense -LogLevel WarningsInfo -LogPath c:\dism.log


Path          :
Online        : True
RestartNeeded : False
```

**Comment by agiacomolli** [comment]
Yes, I've tested it on other Windows versions, and it works as expected. I got the same results as you. I believe the issue is likely to occur on that specific build as you mentioned. To be honest I found it wasn't working by chance.

On that specific Windows build (21966), the command gets stuck when using either dism or the cmdlet. Here is the dism output, which never completes:

```powershell
C:\> dism.exe /LogPath:"c:\dism.log" /online /Add-ProvisionedAppxPackage /PackagePath:"C:\Microsoft.UI.Xaml.2.8.x64.appx" /SkipLicense

Deployment Image Servicing and Management tool
Version: 10.0.21996.1

Image Version: 10.0.21996.1
```

The logs in `dism.log` show the same output I posted earlier.

Just a bit more context, upon installation, our software automatically provisions the required dependencies, including WinUI. However we cannot control which Windows version the user installs our software on, which is why I asked you about supported Windows builds.

Anyway, I think it's fine to add a manual check to exclude this build on our side. This way, we can fail the install with a useful error message.
